js/main-pin.js

```diff
@@ -45,56 +62,33 @@
     }
   };
 
-  var onMainPinMouseDown = function (evt) {
+  var onMainPinMouseDown = function () {
     handleChange(getMainPinCoords(MainPinSize.HEIGHT));
+  };
 
+  var onClickPreventDefault = function (evt) {
     evt.preventDefault();
-    var startCoordinates = {
-      x: evt.clientX,
-      y: evt.clientY
-    };
-
-    var dragged = false;
-
-    var onMouseMove = function (moveEvt) {
-      moveEvt.preventDefault();
-
-      dragged = true;
-
-      var shift = {
-        x: startCoordinates.x - moveEvt.clientX,
-        y: startCoordinates.y - moveEvt.clientY
-      };
-
-      startCoordinates = {
-        x: moveEvt.clientX,
-        y: moveEvt.clientY
-      };
-
-      mainPin.style.top =
-      mainPin.offsetTop - shift.y + 'px';
-      mainPin.style.left =
-      mainPin.offsetLeft - shift.x + 'px';
-    };
+  };
+  var onStartMove = function () {
+    mainPin.addEventListener('click', onClickPreventDefault, {once: true});
+  };
 
-    var onMouseUp = function (upEvt) {
-      upEvt.preventDefault();
+  var onMove = function (evt) {
```

Comment:
Без риска:
```suggestion
  var onMainPinMove = function (evt) {
```
[`Д 4`](https://up.htmlacademy.ru/javascript/19/criteries#d4)
> Из названия обработчика события и функции-коллбэка следует, что это обработчик